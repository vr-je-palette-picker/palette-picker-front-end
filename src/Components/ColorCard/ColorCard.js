import React, { Component } from 'react';
import './ColorCard.scss'

class ColorCard extends Component {
  constructor({ hex }) {
    super();
    this.state = {
      selected: false,
      hex: hex
    }
  }

  render() {
    return (
    <div>
      <h4 style={{backgroundColor: this.state.hex}}>{ this.state.hex }</h4>
    </div>
    )
  }
}

export default ColorCard;
