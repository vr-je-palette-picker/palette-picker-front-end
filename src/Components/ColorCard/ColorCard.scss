.ColorCard {
  display: flex;
  justify-content: space-evenly;
  height: 85vh;
  width: 20vw;
}

.ColorCard__footer {
  align-content: space-between;
  align-self: flex-end;
  display: flex;
  flex-direction: row;
  margin: 0px 18px 10px 0px;
}

.ColorCard__h4 {
  font-size: 45px;
  margin: 0;
}

.ColorCard__lock--icon {
  height: 40px;
  margin-right: 12px;
  transform: translateX(-50%);
  transform: translateY(25%);
  width: 40px;
}