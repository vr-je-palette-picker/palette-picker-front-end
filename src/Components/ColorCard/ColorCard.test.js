import React from 'react';
import { shallow } from 'enzyme';
import ColorCard from './ColorCard';

describe('ColorCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ColorCard />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})