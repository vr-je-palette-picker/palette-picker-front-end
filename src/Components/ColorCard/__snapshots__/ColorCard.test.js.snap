// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard should match the snapshot 1`] = `
<div
  className="ColorCard"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <h4
    className="ColorCard__h4"
  />
  <MdCheckBoxOutlineBlank
    onClick={[Function]}
  />
</div>
`;
